### DISCUSSION TITLE
Editor de text enriquit per al corrector

### DISCUSSION THREAD

**Discussion by xavivars** [Softcatala/web-2015#258] [Idees]
Quines funcionalitats mínimes pensem que ha de tindre el corrector per a considerar que podem posar-lo en producció?

**Comment by jmontane** [comment]
Considero imprescindible:
- [ ] Poder definir de forma fàcil les formes verbals que usarà el corrector (centrals, valencianes, balears)
- [ ] Opcions addicionals de correcció (equivalent a «Més opcions» actual)
- [ ] Format: negreta, cursiva, subratllat, llistes numerades i sense numerar, encapçalaments, enllaços

No és imprescindible, però seria interessant:
- [ ] Exportació i importació en format .docx i .odt
- [ ] Exportació i importació en format Markdown
- [ ] Format: blocs de codi i citacions

**Comment by pereorga** [comment]
Per mi el format no és imprescindible, si treballa en text pla (i no embolica el format al copiar i enganxar) ja hi hauria molt de guanyat. 

Trobo més interessant integrar la funcionalitat de sinònims, per exemple. Possibilitat d'aplicar format i exportar a diferents formats i fitxers són coses que pot anar bé tenir, especialment si no compliquen ni l'ús i ni la implementació.

**Comment by xavivars** [comment]
No per a la primera versió, però estaria bé afegir funcionalitat d'introducció d'emojis, tal com té per exemple GitHub quan fas una cosa com :smile:  `:smile:`.

El primer pas seria aconseguir una llista com aquesta traduïda al català
https://gitlab.com/-/emojis/2/emojis.json

(si optem per TipTap, podem inspirar-nos en https://gitlab.com/gitlab-org/gitlab/-/blob/master/app/assets/javascripts/content_editor/extensions/emoji.js)

**Reply by jmontane** [reply]
Et refereixes a traduir el nom/descripció de l'emoji https://github.com/unicode-org/cldr-json/blob/main/cldr-json/cldr-annotations-full/annotations/ca/annotations.json? O a l'identificador que permet introduir-lo?

**Reply by xavivars** [reply]
La veritat... no ho sé :rofl: 

Supose que l'identificador, no? A l'estil de com funciona el telegram...

**Comment by jordimas** [comment]
Requisits una mica més tècnics:
- [ ] Que sigui compatible amb tots els navegadors pels quals tenim un transit de més del 1%. El corrector amb 39,7M de visites un 1% voldria dir no donar servei a 397.000 visites anuals
- [ ] Que funcioni bé amb mòbil i escriptori
- [ ] Que permeti enganxar i pegar entre el navegador i les aplicacions (com ara)

**Reply by xavivars** [reply]
Respecte als navegadors: Mirant les dades (30 últims dies), sembla que tenim un públic amb els dispositius prou actualitzats. Amb les 3 últimes versions de Chromium (Chrome + Edge), la última major de Safari (16) i les dues últimes de Firefox ho tenim gairebé tot cobert.

![image](https://user-images.githubusercontent.com/134083/226994296-8a9ee8e9-3fd1-423f-ac57-b8f8371440f7.png)

![image](https://user-images.githubusercontent.com/134083/226994607-b72374f8-2b2b-4a4d-912c-d05206d41984.png)
